eventActions/accountabilityActions.js

```diff
@@ -8,37 +20,50 @@ class accountabilityActions {
 		if(reaction.message.channel.id == config.channels.accountability
             && reaction._emoji.name == config.emotes.pinMessage) {
 
-			// Pin the message
 			var sentMessage = reaction.message;
 			var currentChannel = sentMessage.channel;
 
 			// Make sure a user is pinning their own message
-			if(user.id != sentMessage.member.id) return;
-			currentChannel.send('Hey, ' + user.username + ', I\'ve pinned your message for you as requested!');
+			if(user.id != sentMessage.author.id) return;
 
-			// Get the pinned messages within a channel
-			await currentChannel.fetchPinnedMessages().then(fetchedPins => {
+			await currentChannel.fetchPinnedMessages().then(fetchedPins =>{
 
-				// Check to see if they already have pinned messages
-				var pinMsgIterator = fetchedPins.values();
-				var existingMessageCount = 0;
+				// If the pushpin reaction from the bot does not exist, pin the message
+				if(!isMessagePinnedAtAll(sentMessage, fetchedPins)){
+					// Pin the message
+					var existingMessageCount = 0;
 
-				for (var i = 0; i < fetchedPins.size; i++){
-					var msgVal = pinMsgIterator.next().value;
-					if(msgVal.author.id === user.id){
-						existingMessageCount++;
+					// Let the Discord message inform a user something was pinned
+					// currentChannel.send('Hey, ' + user.username + ', I\'ve pinned your message for you as requested!');
```

Comment:
Remove uncommented code

Reply:
The more I look at this the more I think it means the code that was commented out

Done!